Automattic/wp-calypso
# Stepper: Apply extended props to `calypso_signup_step_start`

Part of addressing https://github.com/Automattic/wp-calypso/issues/94108
Branched off https://github.com/Automattic/wp-calypso/pull/93941

## Proposed Changes

* Adds `STEPPER_TRACKS_EVENT_SIGNUP_STEP_START`
* Implements props extension for `calypso_signup_step_start` event
* Some additional refactors to `useStepRouteTracking`. Removes `recordStepStart` method - this was a minimal abstraction that's only relevant to `useStepRouteTracking` (attempting to even reuse this outside will probably cause tracking inconsistencies)

DO NOT MERGE - depended on ongoing discussions in https://github.com/Automattic/wp-calypso/issues/94175 (similar/same problem)

## Why are these changes being made?


* Allow flows to extend the default props tracked by Stepper for all the major/framework events

## Testing Instructions



* test `/setup/onboarding`
  * ensure tracking of `calypso_signup_step_start` when walking through steps in Stepper works like before
  * should fire once per step render (that's not caused by a change in design, intent, or variant)

## Pre-merge Checklist
